src/toil/jobStores/fileJobStore.py

```diff
@@ -565,19 +589,30 @@ def _getSharedFilePath(self, sharedFileName):
         return os.path.join(self.sharedFilesDir, sharedFileName)
 
     @contextmanager
-    def writeSharedFileStream(self, sharedFileName, isProtected=None):
+    def writeSharedFileStream(self, sharedFileName, isProtected=None, mode='b', encoding=None, errors=None):
         # the isProtected parameter has no effect on the fileStore
         self._requireValidSharedFileName(sharedFileName)
         with AtomicFileCreate(self._getSharedFilePath(sharedFileName)) as tmpSharedFilePath:
-            with open(tmpSharedFilePath, 'wb') as f:
-                yield f
+            if mode == 'b':
+                with open(tmpSharedFilePath, 'wb', errors=errors) as f:
+                    yield f
+            elif mode == 't':
+                with open(tmpSharedFilePath, 'w', encoding=encoding, errors=errors) as f:
+                    yield f
 
     @contextmanager
-    def readSharedFileStream(self, sharedFileName):
+    def readSharedFileStream(self, sharedFileName, mode='b', encoding=None, errors=None):
         self._requireValidSharedFileName(sharedFileName)
         try:
-            with open(self._getSharedFilePath(sharedFileName), 'rb') as f:
-                yield f
+            if mode == 'b':
+                with open(self._getSharedFilePath(sharedFileName), 'rb', errors=errors) as f:
+                    yield f
+
+            elif mode == 't':
+                with open(self._getSharedFilePath(sharedFileName), 'r', encoding=encoding,
+                        errors=errors) as f:
+                    yield f
+
```

Comment:
You have a lot of copies of this same sort of structure; could you condense this and maybe put the conditional inside the arguments?

```
                with open(self._getSharedFilePath(sharedFileName), 'r' + (mode if mode != 't' else ''), encoding=encoding,
                        errors=errors) as f:
```

Or if `open` will take `t` you could do:

```
                with open(self._getSharedFilePath(sharedFileName), 'r' + mode, encoding=encoding,
                        errors=errors) as f:
```

Then you would be passing `errors` and `encoding` along even in binary mode, but there would be a good reason for it.